webui/src/lib/components/repository/changes.jsx

```diff
@@ -91,6 +95,43 @@ export const TreeItem = ({ entry, repo, reference, leftDiffRefID, rightDiffRefID
         return {results:resultsState.results, pagination: pagination}
     }, [repo.id, reference.id, internalRefresh, afterUpdated, entry.path, delimiter, dirExpanded])
 
+    let isDeltaTableRootEntry = false
+    if (changesTreeType === ChangesTreeType.DeltaLakeTableChanges) {
+        // The delta tables diff container only shows prefixes.
+        if (entry.path_type === "object") {
+            return
+        }
+
+        // The delta tables diff container only shows prefixes that are either delta tables or prefixes that can
+        // potentially lead to delta tables. The later are prefixes that have prefixes under them as opposed to only
+        // have objects.
+
+        // Tree items that represent prefixes are always of entry.type = prefix_changed and the actual diff type is
+        // presented at the object level. Therefore, in case of delta tables that were added or removed we don't know
+        // under which of the diff refs the prefix is expected to be listed and therefore we list both and take the one
+        // that returned results.
+        const listRight = useAPI(() => objects.list(repo.id, rightDiffRefID, entry.path), []);
+        const listLeft = useAPI(() => objects.list(repo.id, leftDiffRefID, entry.path), []);
+        let listPrefix = listRight.response !== null && listRight.response.results.length !== 0 ? listRight : listLeft
+        if (listPrefix.response !== null && listPrefix.response.results.length !== 0) {
+            let listedItems = listPrefix.response.results;
+            let isPotentialRouteToDeltaTable = false;
+            for (let i = 0; i < listedItems.length; i++) {
+                let item = listedItems[i];
+                if (item.path_type === "common_prefix") {
+                    isPotentialRouteToDeltaTable = true;
+                }
+                if (item.path.includes("_delta_log/")) {
+                    isDeltaTableRootEntry = true;
+                    break;
+                }
+            }
+            if (!isPotentialRouteToDeltaTable) {
```

Comment:
This behavior of not showing directories that have only objects under them, is strange.
If I have two directories, none of which contains any Delta table at any level, I may end up showing just one of them because one has more directories under it and the other doesn't. I would prefer a unified behavior where we present all directories.

Reply:
hmm.. I see, but I'm not sure that showing directories that, when expended, show nothing is clear. Maybe what I can do is remove the expansion option from the prefixes I currently removed. I mean, show all prefixes but don't allow expending prefixes with only objects under them, WDYT? 

Will be happy to get more opinions on this one - @Jonathan-Rosenberg @ozkatz which options is clearer:
1. when in delta lake tables view, show all prefixes, for prefixes that have only objects under them - don't allow expanding. 
2. Only show prefixes that have prefixes under them and therefore can potentially lead to a delta lake table root.